src/sentry/features/manager.py

```diff
@@ -138,24 +138,40 @@ class FeatureManager(RegisteredFeatureManager):
     def __init__(self) -> None:
         super().__init__()
         self._feature_registry: MutableMapping[str, type[Feature]] = {}
+        # Deprecated: Remove entity_features once flagr has been removed.
         self.entity_features: MutableSet[str] = set()
+        self.exposed_features: MutableSet[str] = set()
         self.option_features: MutableSet[str] = set()
         self.flagpole_features: MutableSet[str] = set()
         self._entity_handler: FeatureHandler | None = None
 
-    def all(self, feature_type: type[Feature] = Feature) -> Mapping[str, type[Feature]]:
+    def all(
+        self, feature_type: type[Feature] = Feature, api_expose: bool = True
+    ) -> Mapping[str, type[Feature]]:
         """
         Get a mapping of feature name -> feature class, optionally specific to a
         particular feature type.
+
+        :param feature_type: The feature class you want to filter by. eg. (OrganizationFeature | ProjectFeature | SystemFeature)
+        :param api_expose: Only include features that were registered with `api_expose`.
         """
-        return {k: v for k, v in self._feature_registry.items() if issubclass(v, feature_type)}
+        return {
+            name: feature
+            for name, feature in self._feature_registry.items()
+            if issubclass(feature, feature_type)
+            and (
+                (api_expose and name in self.exposed_features)
+                or (not api_expose and name not in self.exposed_features)
+            )
```

Comment:
Oh I might have misunderstood the point of `api_expose` here. I thought we wanted to return only api expose when true, and everything when false. In that case I think the `None` makes sense for all?

Reply:
I had been thinking about as a filter condition that had an on/off/undefined state. However, we don't really have any scenarios right now where we need all the features that are hidden from the API responses 🤔